pkg/endpoint/endpoint.go

```diff
@@ -2312,7 +2312,7 @@ func (e *Endpoint) WaitForFirstRegeneration(ctx context.Context) error {
 		}
 
 		if ctx.Err() != nil {
-			return fmt.Errorf("timeout while waiting for initial endpoint generation to complete")
+			return fmt.Errorf("timeout while waiting for initial endpoint generation to complete: %w", ctx.Err())
```

Comment:
I don't think this will get us the actual underlying error. It will probably just show `deadline exceeded` or some other `context.Context` standard error msg when the context expires.

Reply:
Oh, I thought that was the actual underlying error message. Can we get something more?

Reply:
Unfortunately not, see godoc on this method. It's only possible to get the error from whatever is return an error, not from the context.

```go
// If Done is not yet closed, Err returns nil.
// If Done is closed, Err returns a non-nil error explaining why:
// Canceled if the context was canceled
// or DeadlineExceeded if the context's deadline passed.
// After Err returns a non-nil error, successive calls to Err return the same error.
Err() error
```

Reply:
OK. But then knowing if cancelled vs. deadline exceeded is still useful information.